### PR TITLE
fixup! reworked and fixed UART driver

### PR DESCRIPTION
**Pull request by haukepetersen**
- the driver does not add extra ROM usage for 'untouched' CPUs (nrf51, nrf52832)
- the driver is more efficient for the `nrf52840`:
  - using the `ENDRX` interrupt
  - remove `assert` from interrupt routine
  - use `SHORTS_ENDRX_STARTRX` short (RX reg is double buffered...)
- fixed broken `UART_1_ISR` define in board config -> needs to be mapped to `isr_uarte1`
- structured the code a little different to make it easier to read

Tested with `nrf52dk` and `nrf52840dk`.

### CODE REVIEW

File: cpu/nrf5x_common/periph/uart.c [hunk 1]
```diff
@@ -111,66 +104,84 @@ int uart_init(uart_t uart, uint32_t baudrate, uart_rx_cb_t rx_cb, void *arg)
     gpio_init(UART_PIN_TX, GPIO_OUT);
     PSEL_TXD = UART_PIN_TX;
 
+#ifdef CPU_MODEL_NRF52840XXAA
     /* enable HW-flow control if defined */
     if (UART_HWFLOWCTRL) {
+        if (uart == 1) {
```
Comment (aabadie) [inline cpu/nrf5x_common/periph/uart.c]: leftover from debugging session ?

File: cpu/nrf52/include/periph_cpu.h [hunk 2]
```diff
@@ -43,9 +43,6 @@ extern "C" {
 #define SPI_MOSISEL         (dev(bus)->PSEL.MOSI)
 #define SPI_MISOSEL         (dev(bus)->PSEL.MISO)
 #ifndef CPU_MODEL_NRF52840XXAA
-#define UART_PIN_RTS        (GPIO_UNDEF)
```
Comment (aabadie) [inline cpu/nrf52/include/periph_cpu.h:46]: IIRC, this was required for nrf51/nrf52832 boards to build nicely if the defines don't exist in the board configuration.

**Review by aabadie — CHANGES_REQUESTED** [review]
Just a couple of comments. Thanks for improving!

File: cpu/nrf5x_common/periph/uart.c (hunk 1 above)
Comment (haukepetersen) [reply cpu/nrf5x_common/periph/uart.c]: yes :-)

File: cpu/nrf52/include/periph_cpu.h (hunk 2 above)
Comment (haukepetersen) [reply cpu/nrf52/include/periph_cpu.h:46]: but its not needed anymore with the restructured code...

**Comment by haukepetersen** [comment]
removed the debug printf....

File: cpu/nrf52/include/periph_cpu.h (hunk 2 above)
Comment (aabadie) [reply cpu/nrf52/include/periph_cpu.h:46]: `UART_HWFLOWCTRL` is still required I think.

Comment (aabadie) [reply cpu/nrf52/include/periph_cpu.h:46]: ok, I checked, and that's fine.

**Pull request merged**